addons/pos_mollie/static/src/app/payment_mollie.js

```diff
@@ -0,0 +1,176 @@
+        const line = order.paymentlines.find((paymentLine) => paymentLine.cid === cid);
+        return {
+            'description': order.name,
+            'pos_reference': order.uid,
+            'currency': this.pos.currency.name,
+            'amount': line.amount,
+            'session_id': this.pos.pos_session.id,
+            'config_id': this.pos.pos_session.config_id[0],
+        }
+    }
+
+    _handle_odoo_connection_failure(data = {}) {
+        const line = this.get_pending_mollie_line();
+        if (line) {
+            line.set_payment_status("retry");
+        }
+        this._show_error(
+            _t("Could not connect to the Odoo server, please check your internet connection and try again.")
+        );
+        return Promise.reject(data);
+    }
+
+    /**
+     * This method handles the response that comes from Mollie
+     * when we first make a request to pay.
+     */
+    _mollie_handle_response(response) {
+
+        const line = this.get_pending_mollie_line();
+
+        if (response.status != 'open') {
+            this._show_error(response.detail);
+            line.set_payment_status('retry');
+            return Promise.resolve();
+        }
+        if (response.id) {
+            line.transaction_id = response.id;
+        }
+        line.set_payment_status('waitingCard');
+        return this.waitForPaymentConfirmation();
+
+    }
+
+    waitForPaymentConfirmation() {
+        return new Promise((resolve) => {
+            this.paymentLineResolvers[this.get_pending_mollie_line().cid] = resolve;
+        });
+    }
+
+    /**
+     * This method is called from pos_bus when the payment
+     * confirmation from Mollie is received via the webhook.
+     */
+    async handleMollieStatusResponse(response) {
+
+        const line = this.get_pending_mollie_line();
+        if (!response) {
+            this._handle_odoo_connection_failure();
+            return;
+        }
+
+        if (line.transaction_id != response.id) {
+            return;
+        }
+
+        if (response.status == 'paid') {
+            this._resolvePaymentStatus(true);
+        } else if (['expired', 'canceled', 'failed'].includes(response.status)) {
+            this._resolvePaymentStatus(false);
+        }
+    }
+
+    _resolvePaymentStatus(state) {
+        const line = this.get_pending_mollie_line();
+        const resolver = this.paymentLineResolvers?.[line.cid];
+        if (resolver) {
+            resolver(state);
+        } else {
+            line.handle_payment_response(state);
+        }
+    }
+
+    // --------------------
+    // HELPER FUNCTIONS
+    // --------------------
+
+    get_pending_mollie_line() {
+        return this.pos.getPendingPaymentLine("mollie");
+    }
+
+    _show_error(msg, title) {
+        if (!title) {
+            title = _t("Mollie Error");
+        }
+        this.env.services.popup.add(ErrorPopup, {
+            title: title,
+            body: msg,
+        });
+    }
+}
```

Comment:
lift this to `PaymentInterface` too

Reply:
This method is available in 2 other pos payment modules, i think better do it in another task.

What do you think ?